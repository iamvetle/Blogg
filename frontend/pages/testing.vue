<template>
  <div class="flex">

    <form @submit.prevent method="post" class="flex flex-col" enctype="multipart/form-data">

      <label for="name">Name:</label>
      <input type="text" name="name" id="name">

      <label for="email">Email</label>
      <input type="email" name="email" id="email">

      <label for="message">Message:</label>
      <textarea name="message" id="" cols="30" rows="10" placeholder="Write a message here"></textarea>

      <label for="file">File:</label>
      <input type="file" name="file" id="file">

      <button type="submit">Submit</button>
    </form>

  </div>
</template>

<script setup lang="ts">
onMounted(() => {

  const form = document.querySelector("form")

  form?.addEventListener("submit", (e: any) => {


    const formData = new FormData(form)

    for (let item of formData) {
      console.log(item[0], item[1])
    }
    
    // ta axios of prÃ¸v fetch her
  })





})

// watchEffect(() => {

// })

</script>

<style scoped></style>