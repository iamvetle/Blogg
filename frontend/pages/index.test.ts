import index from "./index.vue";
import { shallowMount } from "@vue/test-utils";
import { createTestingPinia } from "@pinia/testing";

let wrapper: any;
let pinia: any = createTestingPinia();

// let generalStore;
// let postStore;
// let loggedInUserStore;
// let paginationStore;

const factory = () => {
    return shallowMount(index, {
        global: {
            plugins: [pinia],
            components: {},
            mocks: {},
            stubs: {
                FeedMain:true,
                FeedSidebar:true
            },
        },
        props: {},
        slots: {},
    });
};

describe("Testing the index page", () => {

    beforeEach(() => {
        vi.stubGlobal("definePageMeta", () => {
            return null;
        });
    });

    afterEach(() => {
        if (wrapper) {
            wrapper.unmount();
        }
    });

    test("Should exist", () => {
        wrapper = factory();

        expect(wrapper.exists()).toBe(true);
    });
    test('Should have the FeedMain component', () => {
        wrapper = factory()

        expect(wrapper.findComponent({ name:"FeedMain"}).exists()).toBe(true)
    })
    test('Should have the FeedSidebar component', () => {
        wrapper = factory()

        expect(wrapper.findComponent({ name:"FeedSidebar"}).exists()).toBe(true)
    })
    test('Should match snapshot', () => {
        wrapper = factory()
        expect(wrapper).toMatchSnapshot()
    })
    
});
