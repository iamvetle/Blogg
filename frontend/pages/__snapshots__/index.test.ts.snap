// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the index page > Should match snapshot 1`] = `
"<div class=\\"mt-8 max-w-[1100px] w-full mx-auto px-6 grid grid-cols-10 gap-28\\">
  <div data-test=\\"everything\\" class=\\"col-span-6 mx-auto w-full\\">
    <!-- 
				* No api fetching is here.
				* No url \\"manipulation\\" (local, or api) changing either.
			-->
    <feed-main-stub></feed-main-stub>
  </div>
  <div id=\\"feed-sidebar\\" class=\\"col-span-4 mx-auto w-full\\">
    <feed-sidebar-stub></feed-sidebar-stub>
  </div>
</div>"
`;
